### PR TITLE
Add Build Flags

### PR DESCRIPTION
**Pull request by s9latimm**
Support ASan and UBSan with CI and add debug warning flags.

- [x] `-Wall -Wextra -Wno-unused-parameter` as default for `Debug`
- [x] Define `VARA_FEATURE_SANITIZE`
  - [x] `-fsanitize=address,undefined`
  - [x] Fix build failure with bindings / CPython
- [x] Improve readability of workflow `cmake` invokation

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -33,11 +33,28 @@ if (${VARA_FEATURE_COLORED_OUTPUT})
   endif ()
 endif ()
 
+option(VARA_FEATURE_SANITIZE "Build vara-features with ASan and UBSan" FALSE)
+if (${VARA_FEATURE_SANITIZE})
+  set(CMAKE_CXX_FLAGS "-fsanitize=address,undefined ${CMAKE_CXX_FLAGS}")
+else ()
+  set(CMAKE_CXX_FLAGS "-fno-sanitize=address,undefined ${CMAKE_CXX_FLAGS}")
```
Comment (vulder) [inline CMakeLists.txt]:
```suggestion
```
We not need to prevent usage should someone add it's own flags.

File: CMakeLists.txt [hunk 2]
```diff
@@ -33,11 +33,28 @@ if (${VARA_FEATURE_COLORED_OUTPUT})
...
 if (${VARA_FEATURE_BUILD_PYTHON_BINDINGS})
+  if (${VARA_FEATURE_SANITIZE})
+    message(FATAL_ERROR "Cannot build python bindings together with sanitizers.")
+  endif ()
   add_compile_options(-fPIC)
 endif ()
 
+# Extra debug warning flags
+if (CMAKE_BUILD_TYPE EQUAL "Debug")
+  add_compile_options(-Wall -Wextra -Wno-unused-parameter -O0)
```
Comment (vulder) [inline CMakeLists.txt]:
```suggestion
  add_compile_options(-Wall -Wextra -Wno-unused-parameter)
```
We should not force O0 here, llvm normal debug ("dev") builds use for example `O2`

File: CMakeLists.txt [hunk 3]
```diff
@@ -33,11 +33,28 @@ if (${VARA_FEATURE_COLORED_OUTPUT})
...
+elseif (CMAKE_BUILD_TYPE EQUAL "Release")
+  add_compile_options(-O3)
```
Comment (vulder) [inline CMakeLists.txt]:
```suggestion
```
I'm not sure but is this necessary? As far as I know `Release` should automatically add -O3, maybe I'm wrong.

**Review by vulder — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 3 above)
Comment (s9latimm) [reply CMakeLists.txt]: You are right, default values are
```
CMAKE_CXX_FLAGS_DEBUG='-g'
CMAKE_CXX_FLAGS_RELEASE='-O3 -DNDEBUG'
```

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -59,23 +59,47 @@ jobs:
...
+          cmake .. \
+            -DCMAKE_BUILD_TYPE=$BUILD_TYPE \
+            -DLLVM_REQUESTED_VERSION=$LLVM_VERSION \
+            -DCMAKE_CXX_COMPILER=$CXX \
+            -DVARA_FEATURE_SANITIZE=False \
+            -DVARA_FEATURE_BUILD_PYTHON_BINDINGS=True \
+            -DPYTHON_EXECUTABLE=$(which python3)
+          cmake --build . -- -j
+
+      - name: Run Tests
```
Comment (vulder) [inline .github/workflows/build.yml]:
```suggestion
      - name: Run Python Binding Tests
```

**Review by vulder — APPROVED** [review]

**Pull request merged**